nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-core/src/main/java/org/nuxeo/connect/client/status/ConnectStatusHolder.java

```diff
@@ -124,4 +125,30 @@ public SubscriptionStatusWrapper getStatus(boolean forceRefresh) {
         return instanceStatus;
     }
 
+    /**
+     * Returns the registration expiration timestamp included in the CLID, or -1 if the CLID cannot be loaded or doesn't
+     * include the expiration timestamp (old v0 format).
+     *
+     * @since 10.2
+     */
+    public long getRegistrationExpirationTimestamp() {
+        LogicalInstanceIdentifier clid = getService().getCLID();
+        if (clid == null) {
+            return -1;
+        }
+        String clid1 = clid.getCLID1();
+        if (clid1.length() == 36) {
+            // no expiration timestamp (old v0 format)
+            return -1;
+        }
+        // check format
+        String[] split = clid1.split("\\.");
+        if (split.length != 3) {
+            // invalid format
+            return -1;
+        }
+        // return expiration timestamp
+        return Long.parseLong(split[1]);
```

Comment:
As a followup commit, could you catch `NumberFormatException` here and return `-1` too? With a test.

Reply:
Is that really needed, knowing that:
1/ NOS is generating the CLID
2/ Changing it by hand will lead to an authentication error when fetching the Connect status earlier in the execution stack```WARN  [ConnectStatusHolder] Cannot authenticate against Nuxeo Online Services
org.nuxeo.connect.connector.ConnectSecurityError: Connect server refused authentication (returned 401)```

Reply:
But by the same logic we wouldn't need the `split.length != 3` test?
Anyway ok it's not important to be that robust against corrupted CLIDs.

Reply:
Yes, right.
:+1: